src/Disks/IO/CachedOnDiskWriteBufferFromFile.cpp

```diff
@@ -245,7 +245,7 @@ void FileSegmentRangeWriter::jumpToPosition(size_t position)
         if (position < current_write_offset)
             throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot jump backwards: {} < {}", position, current_write_offset);
 
-        file_segment.complete(false);
+        file_segments->completeAndPopFront(/*allow_background_download=*/false, /*force_shrink_to_downloaded_size=*/false);
```

Comment:
the only place I'm not sure if we want to force shrink to downloaded size.
Also I don't see the function used anywhere, maybe I simple delete it @kssenii ?